src/neo/Ledger/Blockchain.cs

```diff
@@ -443,7 +443,7 @@ private void Persist(Block block)
                 }
                 snapshot.BlockHashIndex.GetAndChange().Set(block);
                 foreach (IPersistencePlugin plugin in Plugin.PersistencePlugins)
-                    plugin.OnPersist(snapshot, all_application_executed);
+                    plugin.OnPersist(snapshot.Clone(), all_application_executed);
```

Comment:
https://github.com/neo-project/neo-modules/blob/c51b96bb8802f48a14e2677a027bed408a0a5b26/src/StatesDumper/StatesDumper.cs#L65

Some plugins use `GetChangeSet()` method. If you pass the cloned snapshot, the method doesn't work.

Reply:
I merged the 2x change, maybe we need to revert it?

Reply:
Maybe I should change plugin not `neo`. But I should mention this is a risk.
@shargon please revert this change on `neox-2.x`.
Let me close this, think twice and will commit another pr.

Reply:
Use `snapshot.CloneReadOnly()`?

Reply:
Or `snapshot.View()` ?